packages/ragbits-core/src/ragbits/core/vector_stores/qdrant.py

```diff
@@ -149,15 +151,35 @@ async def store(self, entries: list[VectorStoreEntry]) -> None:
                     vectors_config=VectorParams(size=vector_size, distance=self._distance_method),
                 )
 
-            points = (
-                models.PointStruct(
-                    id=str(entry.id),
-                    vector=embeddings[entry.id],
-                    payload=entry.model_dump(exclude_none=True),
-                )
-                for entry in entries
-                if entry.id in embeddings
-            )
+            points = []
+            for entry in entries:
+                if entry.id not in embeddings:
+                    continue
+
+                vector = embeddings[entry.id]
+                payload = entry.model_dump(exclude_none=True)
+
+                if isinstance(vector, SparseVector):
+                    # Store sparse vector in payload for Qdrant
+                    payload["_sparse_vector"] = vector.model_dump()
+                    # Use a placeholder dense vector for Qdrant's vector field
+                    # This is needed because Qdrant requires a vector, but we'll use the sparse one from payload
```

Comment:
Qdrant DO natively support sparse vector. The changes in this file don't make sense - we wouldn't be able to retrieve elements if they all have vectors consisting of a single zero. All of this should be replaced with an implementation using the native Qdrant support for sparse vectors.